core/src/test/java/org/jdbi/v3/core/statement/TestSqlLogger.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.statement;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.rule.H2DatabaseRule;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestSqlLogger
+{
+    private static final String CREATE = "create table foo(bar int primary key not null)";
+    private static final String INSERT = "insert into foo(bar) values(1)";
+    private static final String INSERT_NULL = "insert into foo(bar) values(null)";
+    private static final String INSERT_PREPARED = "insert into foo(bar) values(?)";
+
+    @Rule
+    public H2DatabaseRule dbRule = new H2DatabaseRule();
+
+    private Handle h;
+    private TalkativeSqlLogger logger;
+
+    @Before
+    public void before() {
+        logger = new TalkativeSqlLogger();
+        dbRule.getJdbi().getConfig(SqlStatements.class).setSqlLogger(logger);
+        h = dbRule.getJdbi().open();
+    }
+
+    @After
+    public void after() {
+        h.close();
+    }
+
+    @Test
+    public void testStatement() {
+        h.execute(CREATE);
+        h.createUpdate(INSERT).execute();
+
+        assertThat(logger.getRawSql()).containsExactly(CREATE, CREATE, INSERT, INSERT);
+        assertThat(logger.getTimings()).hasSize(2);
+        assertThat(logger.getTimings()).allMatch(x -> x > 0);
```

Comment:
Heads up: you can chain allMatch() after hasSize() without starting a new statement.